services/core/java/com/android/server/clipboard/ClipboardService.java

```diff
@@ -1472,6 +1492,18 @@ private void showAccessNotificationLocked(String callingPackage, int uid, @UserI
                 clipboard.deviceId) == uid) {
             return;
         }
+
+        // Shows a system notification instead, when clipboard access is blocked.
+        if (isReadBlocked) {
+            String message =
+                    getContext().getString(R.string.notif_clipboard_read_deny_title, callingPackage);
```

Comment:
Resource strings are usually not used for logging due to their complexity: they usually vary with device locale, retrieving them is slow, etc